sources/kafka_pipeline.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict
+
+import confluent_kafka  # type: ignore
+import dlt
+
+try:
+    from .kafka import kafka_consumer  # type: ignore
+except ImportError:
+    from kafka import kafka_consumer
+
+
+def load_from_several_topics() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="kafka_pipeline",
+        destination="postgres",
+        dataset_name="kafka_messages",
+    )
+
+    data = kafka_consumer(["purchases", "books"])
+
+    info = pipeline.run(data)
+    print(info)
+
+
+def load_with_custom_processor() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="kafka_pipeline",
+        destination="postgres",
+        dataset_name="kafka_messages",
+    )
+
+    def custom_msg_processor(msg: confluent_kafka.Message) -> Dict[str, Any]:
+        """Convert the given message before loading into the destination.
+
+        Args:
+            msg (confluent_kafka.Message): Message read from the cluster.
+
+        Returns:
+            dict: converted message.
+        """
+        return {
+            "_kafka": {
+                "topic": msg.topic(),  # required field
+                "key": msg.key().decode("utf-8"),
+                "partition": msg.partition(),
+            },
+            "data": msg.value().decode("utf-8"),
```

Comment:
same as https://github.com/dlt-hub/verified-sources/pull/306/files#r1449034951

this works if the record is a JSON, it's pretty common to use avro with a schema registry, if it's the case you might want to consider to le the final user specify a custom serializer.

here my really rough implementation:

```
def message_value_decoder(*, decoder_type, message_value, deserializer):
    """
    Given the type of decoder it returns the decoded message as dict
    """
    if decoder_type == 'json':
        return json.loads(message_value.decode('utf-8'))
    elif decoder_type == 'json_schema':
        message = io.BytesIO(message_value)
        # exclude the first 5 bytes of the message
        # in case the schema registry is used the first byte is the magic byte and the rest is the schema_id
        # in case of JSON encoding we could use the schema_id just to extra validate the event
        magic, schema_id = unpack('>bI', message.read(5))
        logger.debug(f'Message: magic_byte: {magic}, schema_id: {schema_id}')
        return json.loads(message.read())
    elif decoder_type == 'avro_schema':
        return deserializer.__call__(message_value, None)
    else:
        raise ValueError('Decoder type not supported')
```

when working with avro and confluent schema registery a simple implementation of deserializer is the following:
```
from confluent_kafka.schema_registry.avro import AvroDeserializer
deserializer = AvroDeserializer(schema_registry_client=schema_registry_client)
```